ngraph/core/src/op/proposal.cpp

```diff
@@ -75,6 +75,14 @@ void op::v0::Proposal::validate_and_infer_types()
             image_shape_shape[0],
             ").");
 
+        NODE_VALIDATION_CHECK(this,
+                              class_probs_shape[1] * 2 == class_bbox_deltas_shape[1],
+                              "Anchor number inconsistent between class_probs (",
+                              class_probs_shape[1] / 2,
+                              "), and bbox_deltas (",
+                              class_bbox_deltas_shape[1] / 4,
+                              ").");
```

Comment:
Please, cover all the changes in the shape infer function with several unit tests. They should included a number of cases when input dimensions of various inputs are dynamic.

Reply:
Added several unit tests, if you see some gaps let me know.